fontbe/src/features/properties.rs

```diff
@@ -0,0 +1,329 @@
+//! Properties and constants related to unicode data
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+    sync::Arc,
+};
+
+use fontir::ir::Glyph;
+use icu_properties::{script::ScriptWithExtensionsBorrowed, BidiClass, Script};
+use write_fonts::{
+    read::{tables::gsub::Gsub, ReadError},
+    types::{GlyphId, Tag},
+};
+
+use crate::features::ot_tags::{NEW_SCRIPTS, SCRIPT_ALIASES, SCRIPT_EXCEPTIONS_REVERSED};
+
+use super::ot_tags::{DFLT_SCRIPT, INDIC_SCRIPTS, NEW_SCRIPT_TAGS, SCRIPT_EXCEPTIONS, USE_SCRIPTS};
+
+// SAFETY: we can visually verify that these inputs contain only non-null ASCII bytes
+// (this is the only way we can declare these as constants)
+// TODO: remove this if https://github.com/unicode-org/icu4x/pull/4691 is merged/released
+pub const COMMON_SCRIPT: UnicodeShortName =
+    unsafe { UnicodeShortName::from_bytes_unchecked(*b"Zyyy") };
+pub const INHERITED_SCRIPT: UnicodeShortName =
+    unsafe { UnicodeShortName::from_bytes_unchecked(*b"Zinh") };
+
+static SCRIPT_DATA: ScriptWithExtensionsBorrowed<'static> =
+    icu_properties::script::script_with_extensions();
+
+/// The type used by icu4x for script names
+pub type UnicodeShortName = tinystr::TinyAsciiStr<4>;
```

Comment:
I think we should just define a `ScriptTag([u8; 4])` type (or just reuse `Tag`?) and bypass `TinyAsciiStr`. At least for shaping, we definitely don't want to deal with this intermediate type and I don't think it adds any value.

I'd add a const `new([u8; 4]) -> Self` constructor for this and a `new_checked([u8; 4]) -> Option<Self>` constructor that tries to round trip through the `name_to_enum_mapper`.

This also lets us avoid the unsafe code.

Reply:
thinking on this a bit more, I don't think the checked constructor is even necessary but `fn icu_script(self) -> Option<icu_properties::Script>` might be useful and provides the same validation

Reply:
the reason to prefer this type is because it is used by `icu_propeties`, so that we can do things like query the properties of a unicode value, which returns [`Script`](https://docs.rs/icu_properties/1.4.0/icu_properties/struct.Script.html)s, and then we can convert those to the unicode names via that crate.

Reply:
oof, I just wrote something similar a couple weeks ago: https://github.com/dfrg/fount/blob/731e374eb963d30abb4d0f938bb515cc3de0a823/fontique/src/script.rs 🤷

Reply:
looking at yours, is the `unicode-script` crate necessary? I think that information is available via `icu_properties`?

Reply:
in any case my vote right now is that we punt on this and can figure out a more holistic approach when we get to shaping?.

Reply:
yeah, happy to punt on this for now

I added optional unicode-script support because there are potential users that already depend on that library for scripts.